### PR TITLE
fix(combobox): tabindex for combobox cross button

### PR DESCRIPTION
**Pull request by Budhia**
## Description

The following PR has been raised in the light of the March acessibility issue (https://jira-eng-sjc12.cisco.com/jira/browse/CAX-932?filter=32566)

- Updated the web-pack version
- Added tabindex to the cross button in the combobox. This helps user to navigate to the cross button of the combobox using tab and thus enhances the accessibility.

Please go through the link for more detail

## Screenshots:
**Before** (If applicable):

No focus goes inside the combobox 
- No Focus in the combobox cross button

**After**:

- TabPanel TabIndex Change : Removed that blue border highlight
- Added the focus to the cross button in combobox
Screenshot 2021-03-31 at 4 48 58 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
     "@types/sortablejs": "^1.10.6",
     "@types/webpack": "^4.41.7",
     "@types/webpack-dev-server": "^3.10.0",
-    "@types/webpack-merge": "^4.1.5",
+    "@types/webpack-merge": "^5.0.0",
```
Comment (futurethang) [inline web-components/package.json:93]: @Budhia What's up with the upgrade? Were you hitting issues with your build that should go library wide?

File: web-components/package.json [hunk 2]
```diff
@@ -138,7 +138,7 @@
     "webpack": "^4.42.0",
     "webpack-cli": "^3.3.11",
     "webpack-dev-server": "^3.10.3",
-    "webpack-merge": "^4.2.2",
+    "webpack-merge": "^5.7.3",
```
Comment (futurethang) [inline web-components/package.json:141]: @Budhia What's up with the upgrade? Were you hitting issues with your build that should go library wide?

File: web-components/src/components/tabs/TabPanel.ts [hunk 3]
```diff
@@ -24,7 +24,6 @@ export namespace TabPanel {
     protected firstUpdated(changedProperties: PropertyValues) {
       super.firstUpdated(changedProperties);
       this.setAttribute("role", "tabpanel");
```
Comment (futurethang) [inline web-components/src/components/tabs/TabPanel.ts:26]: Removing this is where I wonder about the A11y impact. A screen reader needs this to verbalize that a user has navigated to the Tab Content. The CSS is supposed to show the gighlight when navigated to with Key nav, but NOT show if a user clicks on inner elements with the mouse. If that is how it currently works, then we should keep it.

File: web-components/src/components/combobox/scss/combobox.scss [hunk 4]
```diff
@@ -64,7 +64,7 @@
       color: var(--md-primary-text-color, $md-gray-05);
       cursor: pointer;
       height: $combobox-search-icon-height;
-      padding-top: calc(#{$combobox-main-indent} / 2);
+      padding: calc(#{$combobox-main-indent} / 2);
```
Comment (futurethang) [inline web-components/src/components/combobox/scss/combobox.scss:67]: I also propose this SCSS diff to make sure we are setting the focus style and not relying on browser default. It requires changes here and in ComboBox's `module.scss`
```bash
@@ -65,6 +65,11 @@
       cursor: pointer;
       height: $combobox-search-icon-height;
       padding: calc(#{$combobox-main-indent} / 2);
+
+      &:focus {
+        @include remove-focus;
+        @include focus-boxshadow(rem-calc(2), var(--md-focus-border-color, $lm-combobox-dropdown-item-focus-bg-color-light));
+      }
     }
 
     &:hover {
@@ -273,7 +278,7 @@ ul {
     outline: none;
   }
 
-  ::-webkit-input-placeholder {
+  ::input-placeholder {
     @include text-dots;
     display: inline-block;
     width: calc(100%);
diff --git a/web-components/src/components/combobox/scss/module.scss b/web-components/src/components/combobox/scss/module.scss
index 082f335d5..17be980e3 100644
--- a/web-components/src/components/combobox/scss/module.scss
+++ b/web-components/src/components/combobox/scss/module.scss
@@ -1,9 +1,10 @@
 @import "@/wc_scss/tools/functions/core";
 
 // Settings
-@import "@/wc_scss/colors/settings-base.scss";
-@import "@/wc_scss/colors/settings-transparencies.scss";
-@import "@/wc_scss/settings/core.scss";
+@import "@/wc_scss/colors/settings-base";
+@import "@/wc_scss/colors/settings-transparencies";
+@import "@/wc_scss/settings/core";
+@import "@/wc_scss/tools/mixins/focus";
 @import "@/wc_scss/settings/list-item.scss";
 
 @import "@/wc_scss/typography/settings";

```

**Review by futurethang — CHANGES_REQUESTED** [review]

File: web-components/package.json (hunk 1 above)
Comment (Budhia) [reply web-components/package.json:93]: Yes. While I was trying to setup the repo the dist:dev command was giving issues related to the webpack version. So upgrading it solved the scenario.

File: web-components/package.json (hunk 2 above)
Comment (Budhia) [reply web-components/package.json:141]: @futurethang Yes. While I was trying to setup the repo the dist:dev command was giving issues related to the webpack version. So upgrading it solved the scenario.

File: web-components/src/components/tabs/TabPanel.ts (hunk 3 above)
Comment (Budhia) [reply web-components/src/components/tabs/TabPanel.ts:26]: Sure will be reverting this particular changes back

**Review by futurethang — APPROVED** [review]

**Pull request merged**